server/api/controllers/service.ts

```diff
@@ -0,0 +1,99 @@
+import { Response } from "express";
+import { HeaderMiddleware } from "../middlewares/header";
+import { ServiceService } from "../services/service";
+
+export class ServiceController {
+  public getServices(req: HeaderMiddleware, res: Response) {
+    res.send(req.user.services);
+  }
+
+  public linkGithub(req: HeaderMiddleware, res: Response) {
+    const { code } = req.body;
+
+    ServiceService.linkGithub(code)
+      .then((accessToken) => {
+        ServiceService.saveService({
+          email: req.user.email,
+          name: "GitHub",
+          token: accessToken,
+        })
+          .then((result) => {
+            return res.send({
+              status: "SAVED_GITHUB_SERVICE",
+              message: result,
+            });
+          })
+          .catch((err) => {
+            return res.status(err.code).send({
+              status: "FAILED_TO_SAVE_GITHUB_SERVICE",
+              message: err.message,
+            });
+          });
+      })
+      .catch((err) => {
+        return res
+          .status(err.code)
+          .send({ status: "FAILED_TO_LINK_GITHUB", message: err.message });
+      });
+  }
+
+  public linkHeroku(req: HeaderMiddleware, res: Response) {
+    const { code } = req.body;
+
+    ServiceService.linkHeroku(code)
+      .then((accessToken) => {
+        ServiceService.saveService({
+          email: req.user.email,
+          name: "Heroku",
+          token: accessToken,
+        })
+          .then((result) => {
+            return res.send({
+              status: "SAVED_HEROKU_SERVICE",
+              message: result,
+            });
+          })
+          .catch((err) => {
+            return res.status(err.code).send({
+              status: "FAILED_TO_SAVE_HEROKU_SERVICE",
+              message: err.message,
+            });
+          });
+      })
+      .catch((err) => {
+        return res
+          .status(err.code)
+          .send({ status: "FAILED_TO_LINK_HEROKU", message: err.message });
+      });
+  }
+
+  public linkAsana(req: HeaderMiddleware, res: Response) {
+    const { code, redirect_uri } = req.body;
```

Comment:
Why do we need the ```redirect_url``` to get an access token? Is it required?

Reply:
Ok, sorry, I just found out that we do need it. Do you want me to send it in through the frontend or will you guys populate env variables? I think it's better I send the ```redirect_url``` to the backend.

Reply:
sure it better through the frontend

Reply:
Cool, pushing it in my next PR